cmd/cleanup-controller/handlers/cleanup/match.go

```diff
@@ -0,0 +1,264 @@
+		}
+	}
+	return multierr.Combine(errs...)
+}
+
+func checkResourceFilter(
+	statement kyvernov1.ResourceFilter,
+	// admissionInfo kyvernov1beta1.RequestInfo,
+	resource unstructured.Unstructured,
+	dynamicConfig []string,
+	namespaceLabels map[string]string,
+) []error {
+	var errs []error
+	// if reflect.DeepEqual(admissionInfo, kyvernov1.RequestInfo{}) {
+	// 	rmr.UserInfo = kyvernov1.UserInfo{}
+	// }
+	// checking if the block is empty
+	if statement.IsEmpty() {
+		errs = append(errs, fmt.Errorf("statement cannot be empty"))
+		return errs
+	}
+	matchErrs := checkResourceDescription(
+		statement.ResourceDescription,
+		/* rmr.UserInfo,*/ /*admissionInfo,*/
+		resource,
+		dynamicConfig,
+		namespaceLabels,
+	)
+	errs = append(errs, matchErrs...)
+	return errs
+}
+
+func checkResourceDescription(
+	conditionBlock kyvernov1.ResourceDescription,
+	resource unstructured.Unstructured,
+	dynamicConfig []string,
+	namespaceLabels map[string]string,
+) []error {
+	var errs []error
+	if len(conditionBlock.Kinds) > 0 {
+		if !checkKind(conditionBlock.Kinds, resource.GetKind(), resource.GroupVersionKind()) {
+			errs = append(errs, fmt.Errorf("kind does not match %v", conditionBlock.Kinds))
+		}
+	}
+	resourceName := resource.GetName()
+	if resourceName == "" {
+		resourceName = resource.GetGenerateName()
+	}
+	if conditionBlock.Name != "" {
+		if !checkName(conditionBlock.Name, resourceName) {
+			errs = append(errs, fmt.Errorf("name does not match"))
+		}
+	}
+	if len(conditionBlock.Names) > 0 {
+		noneMatch := true
+		for i := range conditionBlock.Names {
+			if checkName(conditionBlock.Names[i], resourceName) {
+				noneMatch = false
+				break
+			}
+		}
+		if noneMatch {
+			errs = append(errs, fmt.Errorf("none of the names match"))
+		}
+	}
+	if len(conditionBlock.Namespaces) > 0 {
+		if !checkNameSpace(conditionBlock.Namespaces, resource) {
+			errs = append(errs, fmt.Errorf("namespace does not match"))
+		}
+	}
+	if len(conditionBlock.Annotations) > 0 {
+		if !checkAnnotations(conditionBlock.Annotations, resource.GetAnnotations()) {
+			errs = append(errs, fmt.Errorf("annotations does not match"))
+		}
+	}
+	if conditionBlock.Selector != nil {
+		hasPassed, err := checkSelector(conditionBlock.Selector, resource.GetLabels())
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to parse selector: %v", err))
+		} else {
+			if !hasPassed {
+				errs = append(errs, fmt.Errorf("selector does not match"))
+			}
+		}
+	}
+	if conditionBlock.NamespaceSelector != nil && resource.GetKind() != "Namespace" && resource.GetKind() != "" {
+		hasPassed, err := checkSelector(conditionBlock.NamespaceSelector, namespaceLabels)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to parse namespace selector: %v", err))
+		} else {
+			if !hasPassed {
+				errs = append(errs, fmt.Errorf("namespace selector does not match"))
+			}
+		}
+	}
+	// keys := append(admissionInfo.AdmissionUserInfo.Groups, admissionInfo.AdmissionUserInfo.Username)
```

Comment:
Not sure why commenting out userInfo related logic. If we don't have it supported in this PR, would be good to remove them.